numpyro/distributions/constraints.py

```diff
@@ -243,8 +280,16 @@ def __repr__(self):
     def feasible_like(self, prototype):
         return jax.numpy.broadcast_to(self.lower_bound + 1, jax.numpy.shape(prototype))
 
+    def tree_flatten(self):
+        return (self.lower_bound,), (("lower_bound",), dict())
 
-class _Positive(_GreaterThan, _SingletonConstraint):
+    def __eq__(self, other):
+        return isinstance(other, _GreaterThan) & jnp.array_equal(
```

Comment:
Mmh, I think the use of the "bitwise and" operator as a jax-compatible `and` (that does not try to coerce an abstract boolean to a concrete value) is causing `__eq__` to attempt accessing attribute like `lower_bound` even when `other` is not of type `_GreaterThan`. For some reason this happens during the tests, but it could have happened otherwise, so good the tests caught it!